jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/NodeInstanceImpl.java

```diff
@@ -176,15 +175,7 @@ public final void trigger(NodeInstance from, String type) {
     	}
     	
     	if (from != null) {
-    	    int level = ((org.jbpm.workflow.instance.NodeInstance)from).getLevel();
-    	    ((org.jbpm.workflow.instance.NodeInstanceContainer)getNodeInstanceContainer()).setCurrentLevel(level);
```

Comment:
Setting the level is not longer needed?

Reply:
nope. this is really complex to explain, but this piece of code creates new NodeInstanceImpl under the good so it is not good either.